Mathlib/Data/List/Sort.lean

```diff
@@ -356,6 +356,52 @@ theorem sorted_insertionSort : ∀ l, Sorted r (insertionSort r l)
 
 end TotalAndTransitive
 
+theorem orderedInsert_stable {l c : List α} (a : α) (hl : c <+ l) : c <+ orderedInsert r a l := by
+  induction l generalizing c
+  case nil => simp_all only [sublist_nil, orderedInsert, sublist_cons_self]
```

Comment:
Can you try to minimize the use of `case`? Thanks.

Reply:
Sure :) Out of interest, what is the reasoning behind this?

Reply:
It is just stylistic for the library. If you have to use `case`, then that is an indication that the tactic's syntax needs to be extended.